File: Java/T33/src/Constelacao.java
import java.util.ArrayList;

public class Constelacao {
	private String nome;
	private Estrela[] estrelas;
	private int qtAtual;
	
	public Constelacao(String nome, int maxEstrela) {
		this.nome = nome;
		this.estrelas = new Estrela[maxEstrela];
	}

	public double tempConstelacao() {
		double soma = 0.0;
		for(int e=0; e < qtAtual; e++) {
			soma  += estrelas[e].getTemperatura();
		}
		return soma;
	}
	
	public void adicionar(Estrela e) {
		estrelas[qtAtual++] = e;	
	}
	
	public void infoEstrelas()
	{
		for(Estrela e : estrelas){
			System.out.println("Infos das Estrelas: " + e.getNome() + e.getTemperatura() + e.getCor());
		}		
		
	}
		
	
}

File: Scala/T33/src/main/scala/Ex3/Constelacao.scala
package Ex3

import java.util.ArrayList

class Constelacao(private var nome: String, maxEstrela: Int) {

  private var estrelas: Array[Estrela] = new Array[Estrela](maxEstrela)

  private var qtAtual: Int = _

  def tempConstelacao(): Double = {
    var soma: Double = 0.0
    for (e <- 0 until qtAtual) {
      soma += estrelas(e).getTemperatura
    }
    soma
  }

  def adicionar(e: Estrela): Unit = {
    estrelas({ qtAtual += 1; qtAtual - 1 }) = e
  }

  def infoEstrelas(): Unit = {
    for (e <- estrelas) {
      println(
        "Infos das Estrelas: " + e.getNome + e.getTemperatura +
          e.getCor)
    }
  }

}
